src/DependencyInjection/Configuration.php

```diff
@@ -148,4 +149,21 @@ private function createMockNode()
 
         return $node;
     }
+
+    private function createToleranceNode()
+    {
+        $treeBuilder = new TreeBuilder();
+        $node = $treeBuilder->root('tolerance');
+
+        $node
+            ->canBeEnabled()
+            ->children()
+                ->scalarNode('retry')->defaultValue(2)->end()
```

Comment:
Indeed, as @sroze said, it makes sense to have the middleware in the tolerance repo. Integration with CsaGuzzleBundle would simply be a matter of setting a service with the correct tag, or inject the middleware in one of the clients' configuration.